pkg/policy/policy_controller.go

```diff
@@ -262,15 +262,27 @@ func (pc *policyController) Run(ctx context.Context, workers int) {
 	}
 
 	_, _ = pc.pInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    pc.addPolicy,
-		UpdateFunc: pc.updatePolicy,
-		DeleteFunc: pc.deletePolicy,
+		AddFunc: func(obj interface{}) {
+			pc.addPolicy(ctx, obj)
+		},
+		UpdateFunc: func(old, cur interface{}) {
+			pc.updatePolicy(ctx, old, cur)
+		},
+		DeleteFunc: func(obj interface{}) {
+			pc.deletePolicy(ctx, obj)
+		},
 	})
 
 	_, _ = pc.npInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    pc.addPolicy,
-		UpdateFunc: pc.updatePolicy,
-		DeleteFunc: pc.deletePolicy,
+		AddFunc: func(obj interface{}) {
+			pc.addPolicy(ctx, obj)
+		},
+		UpdateFunc: func(old, cur interface{}) {
+			pc.updatePolicy(ctx, old, cur)
+		},
+		DeleteFunc: func(obj interface{}) {
+			pc.deletePolicy(ctx, obj)
+		},
```

Comment:
this should not be handled while adding, we should handle this during reconcilation